src/editor tweaks.js

```diff
@@ -0,0 +1,451 @@
+/*
+👣
+@file editor tweaks
+@summary Various adjustments to the editor that are too minor for individual plugins.  Each tweak is togglable.
+@license MIT
+@author Violgamba (Jon Heard)
+
+@description
+Various editor tweaks, or adjustments to the editor that are too minor for individual plugins.
```

Comment:
i disagree with the idea that multiple small changes are "too minor" for individual plugins. plugins should be modular and self-contained when possible. afaik the only benefit from combining these is that it's slightly less work to import when you want to use all of them, but that's prioritizing a specific workflow over proper architecture (and it's trivial to combine them into a single plugin after-the-fact for local use if that's better for your specific workflow)

Reply:
I'm ok with separating these into individual plugins if you feel strongly about it, but I put them together because:
1) didn't want to spam bipsi-hacks list with a bunch of minor tweaks.  If your ok with that then I suppose its not an issue.
2) I figured it'd be more user friendly this way.  "slightly less work" is underselling it I think.  Downloading many plugins and adding them to each new project isn't nothing, especially if you're making lots of little projects as you learn and play.  If the benefit of a plugin is appreciable, but minor (such as fields starting as tags instead of texts) I personally wouldn't bother, and I doubt I'm alone.  I posit that the average bipsi user would value expediency over proper architecture, especially for quality-of-life features, which aren't necessary and don't effect the game itself.  Also, while I like your idea of combining plugins into a single file (I'm definitely using that) it implies that the user is comfortable enough with JavaScript to (a) recognize that it's possible (you wouldn't do that for a dll plugin system) and (b) have a sufficient sense of workflow to consider and implement the adjustment.  I, again, posit that a large part of the target audience is not JavaScript savvy in any meaningful way.

Ok, during my rant, I realized that one of the features, "tileset-export-ordered-by-tile", is not a quality-of-life tweak, but a feature that's either needed nor not.  With that in mind, I will definitely move it out to it's own plugin.

Again, if you feel strongly about this, then I'll comply.  But I bet that the majority of these features would be, at best, appreciated while browsing, but then never downloaded.

Reply:
> didn't want to spam bipsi-hacks list with a bunch of minor tweaks

as i said earlier i may become a blocker if there's too much to review, but i'd still prefer multiple small PRs over fewer large ones

> I figured it'd be more user friendly this way...

i appreciate the points you've made here, but what you're describing is a conflict between power user and beginner behaviour: you want to do many complex things and not be slowed down by the overhead for each piece, so you've introduced a solution which cuts out the repetition, and you're sympathetic to users who would be put off by that repetition. however, you're overlooking many possible combinations of user needs and proficiencies where this system would make things more complex:

- if someone does not know what the editor tweaks are, it's more difficult at a glance to discover them and learn whether any are something they'd benefit from, instead of clearly seeing a 1:1 list of plugins:features
- if someone wants to use one but not all of these tweaks, they need to learn and use the additional config that only exists to simplify other workflows. this configuration is not particularly complex, but it is redundant with the plugin system itself, which for a beginner means learning two things instead of one
  - you mentioned that you see these (minus the tileset order) as quality of life improvements, but as i mentioned i only think the save + hotkeys are _strictly_ improvements: although i can easily see how the others would be useful, i can also imagine how they might disrupt someone else's workflow
- if someone wants to customize one of these tweaks, they need to spend more time learning the code to find which parts of it are actually relevant
- similarly, if someone has no experience with JS, but may be learning through their exposure to the plugin system, they may be intimidated by the scope of the combined multi-tweak plugin instead of encouraged by the relative simplicity of an individual tweak plugin. with bitsy, multiple people learned JS for the first time via bitsy hacks and commented on how the modular nature helped make it approachable, and especially since bipsi uses JS as part of its in-engine scripting, it's reasonable to assume bipsi plugins might encourage someone to learn JS in the same way.

the fact that adding many plugins to a new project can be tedious is a valid concern, but this is a separate quality of life problem with multiple solutions:

- you could combine multiple plugins into one for reuse
- you could fork bipsi and include these features in the editor by default
- you could import your standard plugins into bipsi and export a "template" project for reuse

"import multiple plugins as a single action" is even something that could be suggested as another bipsi editor improvement itself.

Reply:
Tl;dr: You make good points.  I'll split this plugin up and send separate PRs.

> they need to learn and use the additional config that (...) is redundant with the plugin system itself...

Yeah, I guess I can see how this could be considered redundant.  On reflection, This plugin's architecture is a reaction to the current plugin system as much as the minor nature of the tweaks themselves.

> multiple people learned JS for the first time via bitsy hacks and commented on how the modular nature helped make it approachable

True, and rather inspiring.

> adding many plugins to a new project (...) is a separate quality of life problem with multiple solutions:

Good point.  This multi-tweak plugin locks users into a one-plugin solution, but a general solution could be created that is superior, and select-able, rather than enforced.